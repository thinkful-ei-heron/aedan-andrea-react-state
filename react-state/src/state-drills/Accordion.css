body {
    background-color: rgb(34, 34, 34);
}

ul {
    list-style-type: none;
    padding:0;
    width: 50%;
    margin: 50px auto;
}

button {
    width: 100%;
    line-height: 50px;
    margin-bottom: 10px;
    font-size: 30px;
    background-color: rgb(56, 55, 55);
    color: white;
    border:1px solid rgb(3, 216, 181);
}

p {
    text-align: center;
    font-size: 20px;
    color:white;
}