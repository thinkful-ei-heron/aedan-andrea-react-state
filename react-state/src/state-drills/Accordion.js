import React from 'react';
import './Accordion.css';

export default class Accordion extends React.Component {

    state = {
        active: null
    };

    handleClick(index) {
        this.setState({
            active: index
        });
    }

    render () {
        if(!this.props.sections){
            return (
                <ul>
                    <li>

                    </li>
                </ul>
            )
        } else{
            let sectionArray = this.props.sections.map((item, index) => {
                return (
                    <li key = {index}>
                        <button onClick = {() => this.handleClick(index)}>{item.title}</button>
                        {this.state.active === index ? <p>{item.content}</p> : null}
                    </li>
                );
            });
            return (
                <ul>
                    {sectionArray}
                </ul>
            );
        }
    }
}

