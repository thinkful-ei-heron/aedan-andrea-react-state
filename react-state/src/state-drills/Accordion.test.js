import React from 'react';
import Accordion from './Accordion';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Store from '../Store';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Accordian Component', () => {
    it('renders empty li if no prop supplied', () => {
        const tree= renderer
            .create(<Accordion />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('component renders no sectios as active by default', () =>{
        const tree= renderer
        .create(<Accordion sections={Store.sections} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
    it('component opens a clicked section', () => {
        const wrapper = shallow(<Accordion sections={Store.sections} />)
        wrapper.find('button').at(2).simulate('click')
        expect(toJSON(wrapper)).toMatchSnapshot()
    });
    it('only opens last section when multiple sectionc clicked', () => {
        const wrapper = shallow(<Accordion sections={Store.sections} />)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJSON(wrapper)).toMatchSnapshot()
    });
})