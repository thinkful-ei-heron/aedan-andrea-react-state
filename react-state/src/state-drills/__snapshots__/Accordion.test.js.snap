// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordian Component component renders no sectios as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component only opens last section when multiple sectionc clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordian Component renders empty li if no prop supplied 1`] = `
<ul>
  <li />
</ul>
`;
